components/VoteForGift.jsx

```diff
@@ -0,0 +1,136 @@
+import styled from "styled-components";
+import { BsCircle } from "react-icons/bs";
+import { BsCheckCircle } from "react-icons/bs";
+import { FaRegUserCircle } from "react-icons/fa";
+import { useState } from "react";
+import { useRouter } from "next/dist/client/router";
+
+export const VoteForGift = ({ giftVote, onVoteClick }) => {
+  const router = useRouter();
+  const voteId = router.query.voteId;
+
+  const [idOfClickedGift, setIdOfClickedGift] = useState("");
+  const [voterName, setVoterName] = useState("");
+
+  const handleGiftSelection = (giftId) => {
+    setIdOfClickedGift(giftId);
+  };
+
+  const handleNameInput = (event) => {
+    setVoterName(event.target.value);
+  };
+
+  const increaseVoteCount = async (idOfClickedGift, voterName) => {
+    try {
+      const responsePut = await fetch(`/api/giftVote/${voteId}`, {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          voterName: voterName,
+          idOfClickedGift: idOfClickedGift,
+        }),
+      });
+
+      const responseGet = await fetch(`/api/giftVote/${voteId}`);
+      const giftVoteWithUpdatedVoteCount = await responseGet.json();
+      onVoteClick(giftVoteWithUpdatedVoteCount.vote);
+    } catch (error) {
+      console.error(error.message);
+    } finally {
+    }
+  };
+
+  const voteButtonActivity = () => {
+    if (idOfClickedGift === "" && voterName === "") {
+      return true;
+    } else if (idOfClickedGift === "" || voterName === "") {
+      return true;
+    } else {
+      return false;
+    }
+  };
+
+  const rendered = giftVote.map(({ name, category, link, price, id }) => {
+    return (
+      <SelectGiftButton
+        onClick={() => {
+          handleGiftSelection(id);
+        }}
+        key={id}
+      >
+        <ItemGrid>
+          <Li style={{ gridArea: "name" }}>{name}</Li>
+          <Li style={{ gridArea: "category" }}> {category}</Li>
+          <Li style={{ gridArea: "link" }}>{link}</Li>
+          <Li style={{ gridArea: "price" }}>{price}</Li>
+          {idOfClickedGift === id ? (
+            <BsCheckCircle style={{ gridArea: "emptycircle" }} />
+          ) : (
+            <BsCircle style={{ gridArea: "emptycircle" }} />
+          )}
+        </ItemGrid>
+      </SelectGiftButton>
+    );
+  });
+
+  return (
+    <>
+      <div>{rendered}</div>
+      <VoteForm>
+        <label htmlFor="votername" />
+        <FaRegUserCircle
+          style={{
+            alignSelf: "center",
+            fontSize: "1.5rem",
+            marginRight: "1vw",
+            marginLeft: "1vw",
+          }}
+        />
+        <input
+          type="text"
+          id="votername"
+          name="votername"
+          onChange={handleNameInput}
+          value={voterName}
+        />
+        <button
+          disabled={voteButtonActivity()}
```

Comment:
in regards to the comment above:
```suggestion
          disabled={isVoteButtonDisabled}
```